src/Scriban/Parsing/Parser.Expressions.cs

```diff
@@ -660,6 +660,15 @@ private ScriptExpression ParseExpression(ScriptNode parentNode, ScriptExpression
                         break;
                     }
 
+                    if ((!_isScientific) && (leftOperand is ScriptVariableGlobal) && (parentNode is ScriptPipeCall) && (functionCall == null))
```

Comment:
I don't understand this line related to the tests added above. If `leftOperand is ScriptVariableGlobal` is the case, why the tests above involves `string.contains '2'`, doesn't look like the tests are actually testing this fix, or am I missing something?

Reply:
Test that I have added is exactly the same use case as reported in #360 that showed regression. This test makes sure that fix is no applied at all in this test scenario. Previously (in #344) this ``if`` was passing for the last ``x``, and ``x`` was incorrectly changed to ScriptFunctionCall. It was happening because  ``parentNode`` for last ``x`` was incorrectly set to ScriptPipeCall. So this test makes sure that the parent node for the last ``x`` is not of type ScriptPipeCall and fix is not applied.

Reply:
oh ok, I see the `parentNode` changes above.

But then, where is the test to verify that we don't regress on #360 ?
And to understand the original issue, this was about just the AST, the runtime was working correctly right?
Also wondering why restricting on `leftOperand is ScriptVariableGlobal`? (e.g imagine you have something like `1 | a.b` that would not be caught by this change as well?)

Reply:
This is that test, it checks that regression reported in #360 will not happen again.

Yes the original issue is only about changing one node in AST in a very specific situation, when after pipe operator we have function call without any additional parameter.  Runtime works correctly regardless if AST for this case is fixed or not. 
I need to check that restriction later because it should be applied in that case ``1 | a.b``, it is the most important case for me, I have a lot of function invocation written exactly that way.

Reply:
> This is that test, it checks that regression reported in #360 will not happen again.

Sorry I was not clear, not talking about this test in particular. I was referring to a unit test (e.g checking the result of the AST)

> Yes the original issue is only about changing one node in AST in a very specific situation,

Ok, so in that case, that needs to be solved not for a specific situation. Sorry If I didn't pay enough attention to the original PR.

Reply:
Are they any test for AST right now? I do not see any...

Indeed, the fix doesn't work for ``1 | a.b``, but it should, I Will try to come up with a more general solution.

Reply:
> Are they any test for AST right now? I do not see any...

Not any, you are the first for asking such a thing :)
You can add it next to some unit tests we already have.